include/logging/log_core.h

```diff
@@ -210,6 +211,26 @@ extern "C" {
 	)								    \
 	))
 
+#ifdef CONFIG_LOG_MINIMAL
+void log_minimal_hexdump_print(const char *data, size_t size);
+
+#define Z_LOG(_level, fmt, ...) do { \
+		if (Z_LOG_CONST_LEVEL_CHECK(_level)) { \
+			printk(fmt "\n", ##__VA_ARGS__); \
```

Comment:
there is no locking here. Won't it crash if one log interrupts another? I guess that simple console like uart may survive but what about console over cdc acm?

Reply:
it doesn't.